notebook/agentchat_nestedchat_optiguide.ipynb

```diff
@@ -0,0 +1,1436 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "# OptiGuide with Nested Chats in AutoGen \n",
+    "\n",
+    "This is a nested chat re-implementation of OptiGuide referring to: https://github.com/qingyun-wu/autogen-eval/blob/main/application/A4-multi-agent-coding/optiguide_example.ipynb\n",
+    " \n",
+    "\n",
+    "## Requirements\n",
+    "\n",
+    "AutoGen requires `Python>=3.8`. To run this notebook example, please install:\n",
+    "```bash\n",
+    "pip install pyautogen\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import autogen\n",
+    "from typing import Union\n",
+    "\n",
+    "# test Gurobi installation\n",
+    "from gurobipy import GRB\n",
```

Comment:
Most users will not have gurobipy installed.  Probably better to inform them about the dependency in the previous markdown cell to give the user a heads-up.  The same goes for eventlet.  But you could also build the choice into the cell with something like: 
```
try:
    from gurobipy import GRB
except:
    # Ask for permission to install GurobiPy
    user_response = input("Would you like me to install GurobiPy? (yes/no): ")
    if user_response.lower() == 'yes':
    # Install GurobiPy using pip
        os.system('pip install gurobipy')
    else:
        print("GurobiPy installation cancelled.")
```